File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
def solution(n, works):
    import heapq
    
    answer = 0
    max_heap = [-x for x in works]
    heapq.heapify(max_heap)
    
    while n>0:
        now = -1 * heapq.heappop(max_heap)
        if now != 0:
            next_value = -1*(now-1)
        else:
            next_value = 0
        heapq.heappush(max_heap,next_value)
        n -= 1
    
    for i in max_heap:
        answer += (-1*i) * (-1*i)
    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

long long solution(int n, vector<int> works) {
    long long answer = 0;
    priority_queue<int>pq;
    for(auto  work : works){
        pq.push(work);
    }
    
    while(n > 0 && !pq.empty()){
        int now = pq.top();
        pq.pop();
        if(now>0){
            pq.push(now-1);
            n--;
        }
    }
    
    while (!pq.empty()) {
        int now = pq.top();
        pq.pop();
        answer += (long long)now * now;
    }
    
    return answer;
}